controllers/managedclusterview_controller.go

```diff
@@ -0,0 +1,186 @@
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&viewv1beta1.ManagedClusterView{}, builder.WithPredicates(managedClusterViewPredicate, predicate.ResourceVersionChangedPredicate{}, predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.ConfigMap{}).
+		Complete(r)
+}
+
+func hasODFInfoInScope(mc *viewv1beta1.ManagedClusterView) bool {
+	if mc.Spec.Scope.Name == ODFInfoConfigMapName && mc.Spec.Scope.Kind == ConfigMapResourceType {
+		return true
+	}
+	return false
+}
+
+func (r *ManagedClusterViewReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.Logger.With("ManagedClusterView", req.NamespacedName)
+	logger.Info("Reconciling ManagedClusterView")
+
+	var managedClusterView viewv1beta1.ManagedClusterView
+	if err := r.Client.Get(ctx, req.NamespacedName, &managedClusterView); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			logger.Error("Failed to get ManagedClusterView", "error", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := createOrUpdateConfigMap(ctx, r.Client, managedClusterView, r.Logger); err != nil {
+		logger.Error("Failed to create or update ConfigMap for ManagedClusterView", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully reconciled ManagedClusterView", "name", managedClusterView.Name)
+
+	return ctrl.Result{}, nil
+}
+
+func createOrUpdateConfigMap(ctx context.Context, c client.Client, managedClusterView viewv1beta1.ManagedClusterView, logger *slog.Logger) error {
+	logger = logger.With("ManagedClusterView", managedClusterView.Name, "Namespace", managedClusterView.Namespace)
+
+	var resultData map[string]string
+	err := json.Unmarshal(managedClusterView.Status.Result.Raw, &resultData)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal result data: %v", err)
+	}
+
+	reverseLookup := make(map[string]string)
+
+	for key, value := range resultData {
+		if strings.HasSuffix(key, ".config.yaml") {
+			var config StorageClusterConfig
+			err := json.Unmarshal([]byte(value), &config)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal config data for key %s: %v", key, err)
+			}
+
+			providerInfo := config
+			providerInfo.Clients = nil
+			providerInfo.ProviderClusterName = managedClusterView.Namespace
+
+			providerInfoJSON, err := json.Marshal(providerInfo)
+			if err != nil {
+				return fmt.Errorf("failed to marshal provider info: %v", err)
+			}
+
+			for _, client := range config.Clients {
+				reverseLookup[client] = string(providerInfoJSON)
+			}
+		}
+	}
+
+	configMapData := make(map[string]string)
+	for client, info := range reverseLookup {
+		configMapData[client] = info
+	}
+
+	operatorNamespace := os.Getenv("POD_NAMESPACE")
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ClientInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+		Data: configMapData,
```

Comment:
We don't need to set the data here, we can do it inside the CreateOrUpdate directly